hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -482,18 +481,16 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
       throw new RuntimeException("Interrupted during commit", e);
     }
 
-    // Otherwise stick to the {WAREHOUSE_DIR}/{DB_NAME}.db/{TABLE_NAME} path
-    String warehouseLocation = getWarehouseLocation();
-    return String.format(
-        "%s/%s.db/%s",
-        warehouseLocation, tableIdentifier.namespace().levels()[0], tableIdentifier.name());
+    // Otherwise, stick to the {WAREHOUSE_DIR}/{DB_NAME}.db/{TABLE_NAME} path
+    String warehouseLocation = getDatabaseLocation(tableIdentifier.namespace().levels()[0]);
+    return String.format("%s/%s", warehouseLocation, tableIdentifier.name());
   }
 
-  private String getWarehouseLocation() {
+  private String getDatabaseLocation(String warehouseName) {
     String warehouseLocation = conf.get(HiveConf.ConfVars.METASTOREWAREHOUSE.varname);
     Preconditions.checkNotNull(
         warehouseLocation, "Warehouse location is not set: hive.metastore.warehouse.dir=null");
-    return warehouseLocation;
+    return String.format("%s/%s.db", warehouseLocation, warehouseName);
```

Comment:
There might be some unexpected differences between the old path and the new path. One such thing I can think of  is if the configuration contains a `/` character at the end of the string. There might be some other differences also.

We should make sure that we minimise/remove these differences.

Reply:
Thanks for the review, much appreciated. I believe that the trailing slash is already trimmer, but I've added logic to remove the `/` at the end when present, including a test.